src/main/java/com/github/cameltooling/lsp/internal/parser/CamelYamlDSLParser.java

```diff
@@ -220,4 +254,46 @@ private String extractUriFromYamlData(String line) {
 		}
 		return null;
 	}
+
+	private int extractLines(TextDocumentItem textDocumentItem, Position position, String whenToStop,
+							 StringBuilder lines, int pos, boolean cleanNewLine) {
+		return extractLines(textDocumentItem, position, new String[]{whenToStop}, lines, pos, cleanNewLine);
+	}
+
+	private int extractLines(TextDocumentItem textDocumentItem, Position position, String[] whenToStopArray,
+							 StringBuilder lines, int pos, boolean cleanNewLine) {
+		// If it is a multi-line, cleanNewLine decides if we convert it to one very long single line
+		for (Integer lineNo = position.getLine(); lineNo >=0; lineNo--) {
+			String tempLine = parserFileHelperUtil.getLine(textDocumentItem, lineNo);
```

Comment:
This new line is using the method getLine that is deprecated in this PR. Should it use another method? Or we shoudl not deprecate it?

Reply:
Tested with getLines and it worked. So changing it.